build_rules.gradle

```diff
@@ -837,6 +837,179 @@ ext.applyJavaNature = {
+  String optionsConfigFile = System.getProperty('beamITOptions', '')
+
+  // Any additional properties to be appended to benchmark execution command.
+  String extraProperties = System.getProperty('beamExtraProperties', '')
+
+  // Runner which will be used for running the tests. Possible values: dataflow/direct.
+  // PerfKitBenchmarker will have trouble reading 'null' value. It expects empty string if no config file is expected.
+  String runner = System.getProperty('integrationTestRunner', '')
+
+  // Filesystem which will be used for running the tests. Possible values: hdfs.
+  // if not specified runner's local filesystem will be used.
+  String filesystem = System.getProperty('filesystem')
+
+  /* Always required properties: */
+
+  // Pipeline options to be used by the tested pipeline.
+  String integrationTestPipelineOptions = System.getProperty('integrationTestPipelineOptions')
+
+  // Fully qualified name of the test to be run, eg:
+  // 'org.apache.beam.sdks.java.io.jdbc.JdbcIOIT'.
+  String integrationTest = System.getProperty('integrationTest')
+
+  // Relative path to module where the test is, eg. 'sdks/java/io/jdbc.
+  String itModule = System.getProperty('itModule')
+}
+
+// When applied in a module's build.gradle file, this closure provides task for running
+// IO integration tests (manually, without PerfKitBenchmarker).
+ext.enableJavaPerformanceTesting = {
+  println "enableJavaPerformanceTesting with ${it ? "$it" : "default configuration"} for project ${project.name}"
+
+  // Use the implicit it parameter of the closure to handle zero argument or one argument map calls.
+  // See: http://groovy-lang.org/closures.html#implicit-it
+  JavaPerformanceTestConfiguration configuration = it ? it as JavaPerformanceTestConfiguration : new JavaPerformanceTestConfiguration()
+
+  // Task for running integration tests
+  task integrationTest(type: Test) {
+    include "**/*IT.class"
+    systemProperties.beamTestPipelineOptions = configuration.integrationTestPipelineOptions
+  }
+}
+
+// When applied in a module's build.gradle file, this closure adds task providing
+// additional dependencies that might be needed while running integration tests.
+ext.provideIntegrationTestingDependencies = {
+  println "provideIntegrationTestingDependencies with ${it ? "$it" : "default configuration"} for project ${project.name}"
+
+  // Use the implicit it parameter of the closure to handle zero argument or one argument map calls.
+  // See: http://groovy-lang.org/closures.html#implicit-it
+  JavaPerformanceTestConfiguration configuration = it ? it as JavaPerformanceTestConfiguration : new JavaPerformanceTestConfiguration()
+
+  // Add runners needed to run integration tests on
+  task packageIntegrationTests(type: Jar) {
+    def runner = configuration.runner
+    def filesystem = configuration.filesystem
+
+    dependencies {
+
+      /* include dependencies required by runners */
+      if (runner?.contains('dataflow')) {
```

Comment:
<!--new_thread; commit:3165e6e7301ddfb1565b1de36397bfac474d253b; resolved:0-->
Do we have to ignore case ?

Reply:
I think we should not be case sensitive here because it will be easier to use (eg. dataflow and DataFlow is still the same runner).  We should use `equalsIgnoreCase()` instead of contains, because there's no sense to provide multiple runners/filesystems. I'll fix this too, thanks.